build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/jsoncdb/generator/CompilationDatabaseGenerator.java

```diff
@@ -438,15 +440,15 @@ public boolean visit(IResourceProxy proxy) throws CoreException {
 
 	}
 
-	private String findCompilerInPath(ITool tool, IConfiguration config) {
-		if (toolMap.containsKey(tool)) {
-			return "\"" + toolMap.get(tool) + "\""; //$NON-NLS-1$//$NON-NLS-2$
+	private String findCompilerInPath(ITool tool, IConfiguration config, String compilerName) {
```

Comment:
`tool` is now unused. 

```suggestion
	private String findCompilerInPath(IConfiguration config, String compilerName) {
```